app/page/review/index.tsx

```diff
@@ -65,3 +73,48 @@ function ProfileSwitcher(props: {
 
   return children
 }
+
+function PendingRequestConfirmation(props: { request: Request }) {
+  const { request } = props
+
+  const { provider } = useContext(ProviderContext)
+
+  const account = useAccount(request.accountId)
+  const network = useNetwork(request.networkId)
+
+  const [accountActor, setAccountActor] = useState<Account>(null)
+
+  useEffect(() => {
+    async function setupAccountActor() {
+      const actor = await AccountStorageManager.wrap(provider, account)
+      setAccountActor(actor)
+    }
+    setupAccountActor()
+  }, [account.id])
+
+  if (!accountActor) {
+    return
+  }
```

Comment:
How about we encapsulate the logic into a hook  `useAccountWithActor` ? 

```
const { account, accountActor } = useAccountWithActor(request.accountId)
```

Reply:
Yap, I was considering this too. One caution is that the account actor need to be instantiated in async way, which would need another flag like `actorLoaded` to make sure it is ready before using.

This also incurs a situation that some components only need account data and they don't need to wait for actor.

It might be flexible to provide two hooks `useAccount` & `useAccountWithActor`. Or only one hook `useAccount` with actor, and let components ignore the actor if they don't need it.

Would like to know which way is more reasonable to you.

Reply:
um... 
In my opinion, I prefer `useAccountWithActor` for reasons:
- in review page, we can just focus on how to render the page and don't need to care how account actor be instantiated. (and maybe will reuse this custom hook in other place)
- If we're going to add testing to our frontend, it will be easier.

Reply:
Added in [dd5db10](https://github.com/pilagod/waallet-extension/pull/203/commits/dd5db10a83d1cb1c2751c292ed35dbbedeca78cc)